core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1967,27 +1967,31 @@ class WorkspaceService(protected val userInfo: UserInfo,
     * @param billingProject
     * @param billingAccount
     * @param workspaceId
+    * @param rbsHandoutRequestId
     * @param span
     * @return Future[(GoogleProjectId, GoogleProjectNumber)] of the project that we claimed from RBS
     */
   def setupGoogleProject(billingProject: RawlsBillingProject,
                          billingAccount: RawlsBillingAccountName,
                          workspaceId: String,
                          workspaceName: WorkspaceName,
-                         span: Span = null) : Future[(GoogleProjectId, GoogleProjectNumber)] = {
-    val projectPoolType = billingProject.servicePerimeter match {
-      case Some(_) => ProjectPoolType.ExfiltrationControlled
-      case _ => ProjectPoolType.Regular
-    }
-
+                         rbsHandoutRequestId: String,
+                         span: Span = null) : Future[(GoogleProjectId, GoogleProjectNumber)] =
     for {
       googleProjectId <- traceWithParent("getGoogleProjectFromBuffer", span) { _ =>
-        resourceBufferService.getGoogleProjectFromBuffer(projectPoolType, workspaceId)
+        resourceBufferService.getGoogleProjectFromBuffer(
+          if (billingProject.servicePerimeter.isDefined)
+            ProjectPoolType.ExfiltrationControlled else
+            ProjectPoolType.Regular,
+          workspaceId
+        )
       }
 
       _ = logger.info(s"Moving google project ${googleProjectId} to folder.")
       _ <- traceWithParent("maybeMoveGoogleProjectToFolder", span) { _ =>
-        maybeMoveGoogleProjectToFolder(billingProject.servicePerimeter, googleProjectId)
+        billingProject.servicePerimeter.traverse_ {
+          userServiceConstructor(userInfo).moveGoogleProjectToServicePerimeterFolder(_, googleProjectId)
+        }
```

Comment:
nit: I'm not familiar with `traverse_` so it's a bit harder for me to validate this bit of code just from looking at it. 
(without knowing what `traverse_` does, I can't reason whether `moveGoogleProjectToServicePerimeterFolder` will get called or not when `billingProject.servicePerimeter` is empty)

Reply:
`traverse` is quite a common abstraction in FP and has similar behaviour to `map` - here's a short video that introduces the `Traverse` type class in scala that gives a bunch of examples: https://www.youtube.com/watch?v=CSErp1mZodM 
Hint on naming conventions: if you see a function name that ends in `_`, it's likely to ignore the output and return `F[Unit]`, for some functor `F`.